### ISSUE TITLE
Minor comments on joss/paper.md

### ISSUE THREAD

**Issue by sweichwald** [JuliaManifolds/Manopt.jl#96]
Comments on software paper as per 6f74fee as part of https://github.com/openjournals/joss-reviews/issues/3866#event-5537686057.

---

Only minor comments:
- 10: "Nelder Mead" -> "Nelder-Mead"
- 11: "quasi Newton" -> "Quasi-Newton"
- 41: possibly missing an "and" in "framework can"
- 55: missing comma after "manifold"
- 57: maybe mention an example of a manifold with no closed forms in parentheses
- 57: "more generally arbitrary retractions" -> "more general arbitrary retractions"
- 59: maybe rephrase; it seems there are two reasons for the use of retractions: closed form exponential maps are a) unavailable, or b) costly to compute; thus, it may be odd to state that these approximate algorithms are "more efficient" (than what?) especially in cases when, as stated in the preceding sentence, the closed form exponential map may not even be available
- 72ff: it may be helpful to complement each item by mentioning the corresponding names in Manopt.jl explicitly, e.g. one could mention `conjugate_gradient_descent` and `coefficient=DirectionUpdateRule` (`SteepestDirectionUpdateRule`, `ConjugateDescentCoefficient`, `DaiYuanCoefficient`, `FletcherReevesCoefficient`, `HagerZhangCoefficient`, `HeestenesStiefelCoefficient`, `LiuStoreyCoefficient`, and `PolakRibiereCoefficient`) somewhat mimicking the nice layout of the online documentation and providing readers already a more concrete glimpse into Manopt.jl's scope and potential
- 93: typo, redundant "the"
- 95: for readers less familiar with optimization on manifolds, it may be more instructive to drastically reduce `n` and show the Karcher mean in contrast to the Euclidean mean so as to highlight that the Karcher mean still sits on the sphere (e.g. "a meeting half-way between Trondheim and Lausanne would take place in and not under central Europe" :-))?
- 109ff: can the print output precision be meaningfully adjusted or aligned (the latter would, for example, help to visually identify the changes in `F(x)`)
- 122: "There are" -> "The"
- 144: "Bačák, M. (2014)" appears to be a duplicate of 142

**Comment by kellertuer** [comment]
Thanks for the thorough read and these nice comments.
I will quote all comments here, marks those as finished, that I addressed/fixed and add a few comments as well.

> * [x]  10: "Nelder Mead" -> "Nelder-Mead"
> * [x] 11: "quasi Newton" -> "Quasi-Newton"
> * [x] 41: possibly missing an "and" in "framework can"

Since the sentence was a little long already, I opted for a “. They” instead of an “and”

> * [x] 55: missing comma after "manifold"
> * [ ] 57: maybe mention an example of a manifold with no closed forms in parentheses

this might be a little complicated. First, it depends on the metric on chooses on a manifold, and second, I think the [`SphereSymmetricMatrices`](https://juliamanifolds.github.io/Manifolds.jl/stable/manifolds/spheresymmetricmatrices.html) might be en example; I will have to check literature here.

> * [x] 57: "more generally arbitrary retractions" -> "more general arbitrary retractions"
> * [x] 59: maybe rephrase; it seems there are two reasons for the use of retractions: closed form exponential maps are a) unavailable, or b) costly to compute; thus, it may be odd to state that these approximate algorithms are "more efficient" (than what?) especially in cases when, as stated in the preceding sentence, the closed form exponential map may not even be available

I changed “This yields approximate algorithms that are numerically more efficient.”
to 
“Retractions are first order approximations for the exponential map. They provide an alternative to the acceleration free form, if no closed form solution is known. Otherwise, a retraction might also be chosen, when their evaluation is computationally cheaper than to use the exponential map, especially if their approximation error can be stated, see e.g. [@BendokatZimmermann:2021].”

and added the reference.

> * [ ] 72ff: it may be helpful to complement each item by mentioning the corresponding names in Manopt.jl explicitly, e.g. one could mention `conjugate_gradient_descent` and `coefficient=DirectionUpdateRule` (`SteepestDirectionUpdateRule`, `ConjugateDescentCoefficient`, `DaiYuanCoefficient`, `FletcherReevesCoefficient`, `HagerZhangCoefficient`, `HeestenesStiefelCoefficient`, `LiuStoreyCoefficient`, and `PolakRibiereCoefficient`) somewhat mimicking the nice layout of the online documentation and providing readers already a more concrete glimpse into Manopt.jl's scope and potential

Sounds like a very good idea, I will carefully check that and update the list - maybe even linking the names in Manopt.jl to the docs. Thanks for the feedback on the online documentation!

> * [x] 93: typo, redundant "the"
> * [ ] 95: for readers less familiar with optimization on manifolds, it may be more instructive to drastically reduce `n` and show the Karcher mean in contrast to the Euclidean mean so as to highlight that the Karcher mean still sits on the sphere (e.g. "a meeting half-way between Trondheim and Lausanne would take place in and not under central Europe" :-))?

This is a nice idea, I will reduce the number of points. While it might not be visually pleasing to plot the Euclidean mean, comparing the norm of the Euclidean and the Karcher mean is a very good addendum here. I just need a day or two to carefully revise the experiment.

> * [ ] 109ff: can the print output precision be meaningfully adjusted or aligned (the latter would, for example, help to visually identify the changes in `F(x)`)

This would require a little bit of rework in the debug system. Currently `:Change` or `:Cost` generates parts of the debug string where just interpolation (that is `" Text $variable text"` puts the `variable` into a string, which is called interpolation). If I would change this to some format, I can not use just a Symbol anymore but have to find a way to specify a format. Even if I map the Symbol still to some default (then fixed length) format, I would still have to redesign the DebugX functors. See for example [`DebugChange`](https://github.com/JuliaManifolds/Manopt.jl/blob/e6a1aaaad725cbd6ac653e0bb94d129712f9e28b/src/plans/debug_options.jl#L119-L130), where (a) the struct would have to get a field to store the print format (maybe some fprint style?) and line 144ff (where the string is created) would have to be adapted.

This is quite a nice idea and the modularity would make that possible, but it is still quite some work, mainly in the interface (i.e. how the second line in the code block after line 109 would have to look like) and a little bit in the types behind that. this, though, would take some time to be designed and realised. We can open an issue here to keep track but postpone it for later, if that would be ok?

> * [x] 122: "There are" -> "The"
> * [x] 144: "Bačák, M. (2014)" appears to be a duplicate of 142

Excluding the last 4 non-finished changes, the rest is done in commit 7e9334f.

**Comment by sweichwald** [comment]
Re 109ff: Opening an issue to keep track of future development of this feature is certainly ok!

**Comment by kellertuer** [comment]
* I adressed line 57 (manifold without closed from exp) in 83f4af1
* I added the remark on l-72ff – the links to the solvers – in d74d5d2, where the links all point to the most recent version of 0.3.x (just in case one of them breaks in a future version due to restructure of docs/package). Note that the joss branch – when merged – will also trigger a new version (0.3.15 as of now) adding the paper to master ans fixing a few typos I noticed along the was
* Concerning the example (l. 95) I reduced the number of points to 40 and introduced an explanation/comparison to the (Euclidean) mean in R3 and the differences to it. See 8a39c67 for the main part ( 8a39c67 introduces a non-utf8 sigma instead of `π/8`, since the Pandoc OpenJOSS style does not support π in source code blocks).
* thanks for the feedback on the Debug, that is now tracked (including a modelling) in #97.
